source/fundamentals/context.txt

```diff
@@ -1 +1,121 @@
-TODO: Discussion of the ``context`` package
\ No newline at end of file
+=======
+Context
+=======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+The MongoDB Go Driver uses the **context package** from Go's standard
+library. 
+
+The context package consists of the ``Context`` and ``CancelFunc``
+types. The Go Driver uses the both types to simplify the process of
+creating timeout or cancellation tokens for an operation in a blocking
+function. A blocking function is a function that relies on an event,
+such as an input or output, to proceed with its task.
+
+Expiration
+----------
+
+The driver considers a context expired when an operation exceeds its
+timeout or is canceled. The driver checks the context expiration with
+the ``Done()`` function of the ``Context`` type. 
+
+The context is not expired if the function returns a closed channel and
+a timestamp that indicates when the function will cancel the completed
+work. The context is expired if the function returns an open channel
+that you can read from. 
+
+The following sections describe when and how the driver checks for
+expiration. 
+
+Server Selection
+~~~~~~~~~~~~~~~~
+
+The driver might block you if it can't select a server for an operation.
+
+In this scenario, the driver loops until it finds a server to use for the
+operation. After each iteration, the driver returns a server selection
+timeout error if the context expired or the selection process took
+longer than the ``serverSelectionTimeoutMS`` setting. 
+
+For more information on how the driver selects a server, see the
+:ref:`Server Selection Algorithm Server Manual Page
+<replica-set-read-preference-behavior>`. 
```

Comment:
Comment: The reason this ref didn't work is because this repository is missing an intersphinx file that tells the autobuilder where to look for references. For example [here's](https://github.com/mongodb/docs-node/blob/master/config/intersphinx.yaml#L6) the one in the Node driver docs.
@terakilobyte Can probably help set that up. :)

Once it's set up, you won't need to include the link text here -- the title will automatically generate from the referenced page.
```
For more informatino on how the driver selects a server, see :ref:`<replica-set-read-preference-bheavior`.
```
That doesn't necessarily need to hold up this review though.